### PR TITLE
chore: refactored InkAPIPrivateIOCore.h

### PR DESCRIPTION
**Pull request by shikharvashistha**
This commit refactors the header to remove API's mentioned in #6245

Signed-off-by: Shikhar Vashistha <51105234+shikharvashistha@users.noreply.github.com>

### CODE REVIEW

**Comment by brbzull0** [comment]
Hello, Thanks for working on this.
few things to mention:
- Check [here](https://github.com/apache/trafficserver/blob/86790c7a3158898219f807e04c284b172f9a7cfc/src/traffic_server/InkIOCoreAPI.cc#L436) as this needs to be removed too if we are going ahead with this cleanup.
- Haven't check but on the issue it was mentioned that it would be good to run this by sending an email to the dev@ list. You can include the link to this PR .

Thanks again for working on this.

**Comment by bryancall** [comment]
[approve ci]

File: include/ts/InkAPIPrivateIOCore.h [hunk 1]
```diff
@@ -138,64 +138,6 @@ tsapi void TSIOBufferBlockDestroy(TSIOBufferBlock blockp);
 typedef void *INKUDPPacket;
 typedef void *INKUDPacketQueue;
 typedef void *INKUDPConn;
```
Comment (ywkaras) [inline include/ts/InkAPIPrivateIOCore.h:140]: Why not also delete these three INKUDPxxx typedefs?

Comment (shikharvashistha) [reply include/ts/InkAPIPrivateIOCore.h:140]: I think they are being utilized

**Pull request closed without merging**